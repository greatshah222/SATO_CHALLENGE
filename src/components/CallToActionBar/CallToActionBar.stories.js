import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { muiTheme } from 'storybook-addon-material-ui';

import CallToActionBar from './index';
import Button from '../Button';
import theme from '../../theme';

storiesOf('CallToActionBar', module)
  .addDecorator(muiTheme(Object.values(theme)))
  .addWithJSX('with button', () => (
    <CallToActionBar open={true}>
      <Button variant="contained" style={{ background: 'white' }} key={'1'} onClick={action('clicked')}>
        Click me!
      </Button>
    </CallToActionBar>
  ))
  .addWithJSX('with button and text', () => (
    <CallToActionBar
      open={true}
      action={[
        <Button variant="contained" style={{ background: 'white' }} key={'1'} onClick={action('clicked')}>
          Click me!
        </Button>,
      ]}
      message={['Information about clicking ...']}
    />
  ))
  .addWithJSX('transparent', () => (
    <CallToActionBar open={true} transparent>
      <Button color="primary" variant="contained" key={'1'} onClick={action('clicked')}>
        Click me!
      </Button>
    </CallToActionBar>
  ));
