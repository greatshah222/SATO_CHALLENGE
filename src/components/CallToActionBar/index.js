import React from 'react';
import classNames from 'classnames';
import { makeStyles } from '@material-ui/core/styles';
import MuiSnackbar from '@material-ui/core/Snackbar';
import * as tokens from '../../.generated/styles';

const useStyles = makeStyles(theme => ({
  snackbar: {
    width: '100%',
    height: tokens.spacingGargantual,
    background: theme.palette.primary.main,
    bottom: 0,
  },
  content: {
    backgroundColor: 'transparent',
    boxShadow: 'none',
  },
  transparent: {
    background: 'transparent',
    color: tokens.textBlack,
  },
}));

const CallToActionBar = ({ className, transparent, ...props }) => {
  const classes = useStyles();
  return (
    <MuiSnackbar
      className={classNames(classes.snackbar, { [classes.transparent]: transparent }, className)}
      ContentProps={{
        classes: {
          root: classes.content,
        },
      }}
      {...props}
    />
  );
};

export default CallToActionBar;
