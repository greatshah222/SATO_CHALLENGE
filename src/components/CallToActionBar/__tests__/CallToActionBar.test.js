import React from 'react';
import { shallow } from 'enzyme';
import CallToActionBar from '../index';

describe('CallToActionBar', () => {
  it('renders correctly', () => {
    const tree = shallow(<CallToActionBar open={true} />);
    expect(tree).toMatchSnapshot();
  });
});
