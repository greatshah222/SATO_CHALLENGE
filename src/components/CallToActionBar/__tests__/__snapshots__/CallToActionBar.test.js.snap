// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CallToActionBar renders correctly 1`] = `
<WithStyles(ForwardRef(Snackbar))
  ContentProps={
    Object {
      "classes": Object {
        "root": "makeStyles-content-2",
      },
    }
  }
  className="makeStyles-snackbar-1"
  open={true}
/>
`;
